tests/common/dualtor/dual_tor_utils.py

```diff
@@ -596,6 +622,52 @@ def get_crm_nexthop_counter(host):
     return crm_facts['resources']['ipv4_nexthop']['used']
 
 
+@pytest.fixture
+def dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo):
+    """
+    @summary: A helper function for collecting info of dualtor testbed.
+    @param ptfhost: The ptf host fixture
+    @param rand_selected_dut: The randomly selected dut host, will be set as standby ToR
+    @param rand_unselected_dut: The other dut in dualtor testbed, will be set as active ToR
+    @param tbinfo: The tbinfo fixture
+    @return: A dict, can be used as the argument of check_tunnel_balance
+    """
+    active_tor = rand_unselected_dut
+    standby_tor = rand_selected_dut
+    standby_tor_mg_facts = standby_tor.get_extended_minigraph_facts(tbinfo)
+
+    def _get_iface_ip(mg_facts, ifacename):
+        for loopback in mg_facts['minigraph_lo_interfaces']:
+            if loopback['name'] == ifacename and ipaddress.ip_address(loopback['addr']).version == 4:
+                return loopback['addr']
+
+    res = {}
+    res['ptfhost'] = ptfhost
+    res['standby_tor_mac'] = standby_tor.facts['router_mac']
+    vlan_name = standby_tor_mg_facts['minigraph_vlans'].keys()[0]
+    res['vlan_mac'] = standby_tor.get_dut_iface_mac(vlan_name)
+    res['standby_tor_ip'] = _get_iface_ip(standby_tor_mg_facts, 'Loopback0')
+
+    if 't0' in tbinfo["topo"]["name"]:
+        # For mocked dualtor
+        res['active_tor_ip'] = str(ipaddress.ip_address(res['standby_tor_ip']) + 1)
+        # For mocked dualtor, routes to peer switch is static 
+        res['ptf_portchannel_indices'] = get_t1_active_ptf_ports(standby_tor, tbinfo)
+    else:
```

Comment:
Can we explicitly check if `dualtor in tbinf["topo"]["name"]` in case this is accidentally run on a T1 or some other topology?

Reply:
I don't think it will be a problem. The check here is only for distinguishing mocked dualtor (T0) from physical dualtor. Normally, there must be a topology marker at the head of file.
```
pytestmark = [
    pytest.mark.topology('t0')
]
```